### PR TITLE
Remove MPC validation

### PR DESCRIPTION
**Pull request by SurgeCode**

### CODE REVIEW

File: packages/agent-sdk/src/evm/index.ts [hunk 1]
```diff
@@ -59,25 +59,18 @@ export async function validateRequest<
 >(
   req: TRequest,
   // TODO(bh2smith): Use Bitte Wallet's safeSaltNonce as Default.
```
Comment (bh2smith) [inline packages/agent-sdk/src/evm/index.ts:61]: maybe remove this also?
```suggestion
```

File: packages/agent-sdk/src/evm/index.ts [hunk 2]
```diff
@@ -59,25 +59,18 @@ export async function validateRequest<
...
-  safeSaltNonce: string,
   responder?: (data: object, init?: { status?: number }) => TResponse,
 ): Promise<TResponse | null> {
   const createResponse = responder ? responder : fallbackResponder;
   const metadataHeader = req.headers.get("mb-metadata");
   const metadata = JSON.parse(metadataHeader ?? "{}");
   const { accountId, evmAddress } = metadata;
-  if (!accountId || !evmAddress) {
+  if (!accountId && !evmAddress) {
     const error = "Missing accountId or evmAddress in metadata";
```
Comment (bh2smith) [inline packages/agent-sdk/src/evm/index.ts:69]:
```suggestion
    const error = "Missing accountId and evmAddress in metadata";
```

File: packages/agent-sdk/src/evm/index.ts [hunk 3]
```diff
@@ -59,25 +59,18 @@ export async function validateRequest<
...
   // TODO(bh2smith): Use Bitte Wallet's safeSaltNonce as Default.
-  safeSaltNonce: string,
   responder?: (data: object, init?: { status?: number }) => TResponse,
 ): Promise<TResponse | null> {
   const createResponse = responder ? responder : fallbackResponder;
   const metadataHeader = req.headers.get("mb-metadata");
   const metadata = JSON.parse(metadataHeader ?? "{}");
   const { accountId, evmAddress } = metadata;
-  if (!accountId || !evmAddress) {
+  if (!accountId && !evmAddress) {
```
Comment (bh2smith) [inline packages/agent-sdk/src/evm/index.ts:68]: I think we should still be erroring if only one is provided? Why was this change included here?

**Review by bh2smith — APPROVED** [review]
Seems alright. 
Not sure about the change from || to && and how its relevant here. There is a TODO that should be removed and an error message that should be changed based on the or and condition change.

Note that this now exposes us to spam because anyone can call the agent.

**Review by bh2smith — COMMENTED** [review]
Also not that we are not actually validating that the evm Address is an evm address anymore. Before it was indirectly implied by the call to getAddress before.

File: packages/agent-sdk/src/evm/index.ts (hunk 3 above)
Comment (SurgeCode) [reply packages/agent-sdk/src/evm/index.ts:68]: This can be refined but we don't need to have both to create evm transactions anymore

**Comment by SurgeCode** [comment]
> Seems alright. Not sure about the change from || to && and how its relevant here. There is a TODO that should be removed and an error message that should be changed based on the or and condition change.
> 
> Note that this now exposes us to spam because anyone can call the agent.



> Also not that we are not actually validating that the evm Address is an evm address anymore. Before it was indirectly implied by the call to getAddress before.

IMO that should be done by the runtime

**Comment by bh2smith** [comment]
I think you left out some suggestions here.
